src/Features/Core/Portable/DocumentationComments/AbstractDocumentationCommentFormattingService.cs

```diff
@@ -30,8 +30,41 @@ private class FormatterState
             private static TaggedText s_newlinePart = new TaggedText(TextTags.LineBreak, "\r\n");
 
             internal readonly List<TaggedText> Builder = new List<TaggedText>();
+
+            /// <summary>
+            /// Defines the containing lists for the current formatting state. The last item in the list is the
+            /// innermost list.
+            ///
+            /// <list type="bullet">
+            /// <item>
+            /// <term><c>type</c></term>
+            /// <description>The type of list.</description>
+            /// </item>
+            /// <item>
+            /// <term><c>index</c></term>
+            /// <description>The index of the current item in the list.</description>
+            /// </item>
+            /// <item>
+            /// <term><c>renderedItem</c></term>
+            /// <description><see langword="true"/> if the label (a bullet or number) for the current list item has already been rendered; otherwise <see langword="false"/>.</description>
+            /// </item>
+            /// </list>
+            /// </summary>
             private readonly List<(DocumentationCommentListType type, int index, bool renderedItem)> _listStack = new List<(DocumentationCommentListType type, int index, bool renderedItem)>();
+
+            /// <summary>
+            /// The top item of the stack indicates the hyperlink to apply to text rendered at the current location. It
+            /// consists of a navigation <c>target</c> (the destination to navigate to when clicked) and a <c>hint</c>
+            /// (typically shown as a tooltip for the link). This stack is never empty; when no hyperlink applies to the
+            /// current scope, the top item of the stack will be a default tuple instance.
+            /// </summary>
             private readonly Stack<(string target, string hint)> _navigationTargetStack = new Stack<(string target, string hint)>();
```

Comment:
you dont' necessarily have to doc this, but can you help me explain a bit more how this is a stack?  i.e. when do you have a nested stack of these...  it's not really 'clicking' intuitively for me.  Thanks!

Reply:
It might not need to be a stack, but it makes it easy to implement consistently with styles.